### PR TITLE
[oracle] Adding Oracle Active Data Guard support

### PR DESCRIPTION
**Pull request by nenadnoveljic**
### What does this PR do?

Adding Oracle Active Data Guard support.

### Describe how to test/QA your changes

Connect the Agent to a regular (no Data Guard), primary and standby database.

Stop log transportation and check the `transport_lag` and `apply_lag`.

Check that the `hosting_type` tag is initialized for all three databases. Check metrics, query metrics and FQT payloads.

Check that the `database_role` is initialized correctly for primary and standby. The database without the Data Guard configuration shouldn't have the tag.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by lu-zhengda — APPROVED** [review]

File: pkg/collector/corechecks/oracle-dbm/oracle.go [hunk 1]
```diff
@@ -205,6 +224,7 @@ func (c *Check) Run() error {
 	}
 
 	if c.dbmEnabled {
+		//if c.config.QuerySamples.Enabled && c.openMode != "MOUNTED" {
```
Comment (alexandre-normand) [inline pkg/collector/corechecks/oracle-dbm/oracle.go:227]: Delete that commented out line?

File: pkg/collector/corechecks/oracle-dbm/oracle.go [hunk 2]
```diff
@@ -161,6 +171,15 @@ func (c *Check) Run() error {
 	metricIntervalExpired := checkIntervalExpired(&c.metricLastRun, c.config.MetricCollectionInterval)
 
 	if metricIntervalExpired {
+		log.Tracef("%s fixtags here2", c.logPrompt)
```
Comment (alexandre-normand) [inline pkg/collector/corechecks/oracle-dbm/oracle.go:174]: Is that a leftover debug log?

**Review by alexandre-normand — COMMENTED** [review]
Just the two minor polish comments. Otherwise, that's good to go 👍 .

**Review by git-thuerk-done — APPROVED** [review]
Looks good

**Pull request merged**